debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsStreamProducer.java

```diff
@@ -419,10 +428,14 @@ private boolean schemaChanged(List<ReplicationMessage.Column> messageList, Table
             if (column == null) {
                 logger.debug("found new column '{}' present in the server message which is not part of the table metadata; refreshing table schema", columnName);
                 return true;
-            } else if (!schema().isType(column.typeName(), column.jdbcType())) {
-                logger.debug("detected new type for column '{}', old type was '{}', new type is '{}'; refreshing table schema", columnName, column.jdbcType(),
-                            message.getType());
-                return true;
+            } else {
+                final int localType = column.jdbcType();
```

Comment:
Wouldn't we also have to compare scale, precision etc.? How about just comparing on the raw type metadata string (which we'd have to store for that in the `Table` model)?

Reply:
That is how it worked in old code too. I am aware of the issue but I'd prefer to handle it as a separate PR as it will need larger test coverage.

Reply:
Ok, can you file a JIRA issue for it? And - if we don't address it for 0.7 - document that e.g. length changes won't reflect in an updated schema (although I'd strongly prefer to get this right)?

Reply:
https://issues.jboss.org/browse/DBZ-504